staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go

```diff
@@ -45,14 +46,22 @@ func New(config *azclients.ClientConfig) *Client {
 
 // CreateFileShare creates a file share
 func (c *Client) CreateFileShare(resourceGroupName, accountName, name string, sizeGiB int) error {
+	_, err := c.GetFileShare(resourceGroupName, accountName, name)
+	if err == nil {
+		klog.V(2).Infof("file share(%s) under account(%s) rg(%s) already exists", name, accountName, resourceGroupName)
+		return nil
+	} else if err != nil && !strings.Contains(err.Error(), "ShareNotFound") {
```

Comment:
is there any better way to check whether share exists? if error message changed in the future, this function will break

Reply:
if there is no better way other than error msg check, I think keep original code is better although it has a side effect that create on existing file share may change the size.

Reply:
add the condition `result.Response.StatusCode == http.StatusNotFound` to enhance the check